### ISSUE TITLE
Windows libraries for gui

### ISSUE THREAD

**Issue by pazos** [monero-project/monero-gui#1372]
I'm pretty new here in monero-gui so I need some background to understand why we linked the gui against icu on windows. I understand that icu provides unicode support as is useful for *some* qt stuff but I don't understand how the gui itself can use icu libraries directly.

[From qt windows requirements](http://doc.qt.io/qt-5/windows-requirements.html) I read the following:

 > At compile time, the include and lib folders of the ICU installation must be appended to the INCLUDE and LIB environment variables. At run-time, the ICU DLLs need to be found by copying the DLLs to the application folder or by adding the bin folder of the ICU installation to the PATH environment variable.

So I think we add icu to LIBS just to automate the process of copy those dlls to the application via windeployqt. In that case I don't understand why we need to include those to [windeploy_helper.sh](https://github.com/monero-project/monero-gui/blob/master/windeploy_helper.sh).

Also most of the files in the helper script are needed to support msys2 qt (I'm thinking about libpng, libjpeg, zlib...) but we don't need that when using a self built version of qt (which is needed to support different opengl implementations).

So @rbrunner7 @danrmiller @Jaqueeee  and everybody else please give me some points about that issue. If I'm not wrong is just a matter of differenciate between release/debug and release/debug-"static" builds and include two sets of libraries.

**Comment by rbrunner7** [comment]
> Let me see if I understand: ICU libraries are a requisite for boost_locale. The rest of files pulled in windeploy_helper.sh are just to support msys qt.

Well, I did not write that script originally. I just made changes that became known as necessary because it seemed nobody else came around to do them :)

I don't know in detail which file is there for what reason. I currently have no reason to doubt that all are needed, in one way or another, however.

> Also can you confirm that the rest of the files in windeploy_helper.sh aren't generated by windeployqt. That seems weird to me.

I never checked systematically whether any of the files copied are already there, copied already by the Qt deploy mechanism that runs immediately before this script. It can't be however that the script is completely unnecessary and in reality Qt deploy does already the whole job.

And anyway, Monero is a lot of stuff beside the GUI code.
